lib/logstash/codecs/auto_flush.rb

```diff
@@ -11,16 +11,21 @@ def start
     # can't start if pipeline is stopping
     return self if stopped?
     if pending?
-      @task.reset
+      @task.cancel
```

Comment:
Actually, its not a bug. We are supposed to flush a half-built buffer because we never know if a next line even exists.